src/ui/components/form/Vector.tsx

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021 @paritytech/contracts-ui authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { TypeDef } from '@polkadot/types/types';
+import React from 'react';
+import { Button, Buttons } from '../common';
+
+type Props = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Component: React.ComponentType<any>;
+  props: {
+    className: string;
+    value: unknown[];
+    id: string;
+    onChange: (value: unknown[]) => void;
+    nestingNumber: number;
+    type: TypeDef;
+  };
+};
+
+type TitleProps = {
+  count: number;
+  nestingNumber: number;
+  type: string;
+};
+
+const Title = ({ nestingNumber, count, type }: TitleProps) => {
+  if (count == 0) return <>{type}</>;
+  //const vecColor = count == 1 ? 'text-white' : 'text-gray-300';
+  return (
+    <div className={'flex justify-start'}>
+      <div className={count == 1 ? 'dark:text-white text-gray-600' : ''}>{'Vec<'}</div>
+      <Title nestingNumber={nestingNumber} count={count - 1} type={type} />
+      <div className={count == 1 ? 'dark:text-white text-gray-600' : ''}>{'>'}</div>
+    </div>
+  );
+};
+
+export function Vector({
+  Component,
+  props: { value, onChange, nestingNumber, type },
+  props,
+}: Props) {
+  const _rowAdd = () => onChange([...value, '']);
+  const _rowRemove = () => onChange(value.slice(0, -1));
+  const _onChange = (row: number) => (newEntry: unknown) => {
+    const newValue = value.map((entry, index) => (index == row ? newEntry : entry));
+    onChange(newValue);
+  };
+  const isOddNesting = nestingNumber % 2 != 0;
+  return (
+    <div
+      className={`p-4 text-left text-sm ${
+        isOddNesting ? 'dark:bg-gray-900 bg-white' : 'dark:bg-gray-500 bg-gray-100'
```

Comment:
`dark:bg-elevation-1` instead of `dark:bg-gray-500` matches the sidebar color and the mock-up

Reply:
True, looks much better! Done!